backend/app/src/test/kotlin/io/tolgee/slack/SlackIntegrationTest.kt

```diff
@@ -1,40 +1,167 @@
+import java.util.*
 
-class SlackIntegrationTest : AbstractSpringTest(), Logging {
+class SlackIntegrationTest : ProjectAuthControllerTest(), Logging {
   @Autowired
   @MockBean
   lateinit var slackClient: Slack
 
+  lateinit var mockedSlackClient: MockedSlackClient
+
+  @Autowired
+  lateinit var slackMessageService: SavedSlackMessageService
+
+  @BeforeAll
+  fun setUp() {
+    // slackProperties.token = "fakeToken"
+    // tolgeeProperties.slack.token = "fakeToken"
+  }
+
   @Test
-  fun `it mocks slack client`() {
-    val mockedSlackClient = mockSlackClient()
+  fun `sends message to correct channel after translation changed`() {
+    val testData = SlackTestData()
+    testDataService.saveTestData(testData.root)
+    mockedSlackClient = mockSlackClient()
 
-    slackClient.methods("ahhs").chatPostMessage {
-      it.channel("channel")
-      it.text("text")
+    val langTag = testData.projectBuilder.self.baseLanguage?.tag ?: ""
+    loginAsUser(testData.user.username)
+    waitForNotThrowing {
+      modifyTranslationData(testData.projectBuilder.self.id, langTag)
+      mockedSlackClient.chatPostMessageRequests.assert.hasSize(1)
+      val request = mockedSlackClient.chatPostMessageRequests.single()
+      request.channel.assert.isEqualTo(testData.slackConfig.channelId)
     }
+  }
+
+  @Test
+  fun `sends message to correct channel after key added`() {
+    val testData = SlackTestData()
+    testDataService.saveTestData(testData.root)
+    mockedSlackClient = mockSlackClient()
 
-    mockedSlackClient.chatPostMessageRequests.assert.hasSize(1)
-    val request = mockedSlackClient.chatPostMessageRequests.single()
-    request.channel.assert.isEqualTo("channel")
-    request.text.assert.isEqualTo("text")
+    loginAsUser(testData.user.username)
+    waitForNotThrowing {
+      addKeyToProject(testData.projectBuilder.self.id)
+      mockedSlackClient.chatPostMessageRequests.assert.hasSize(1)
+      val request = mockedSlackClient.chatPostMessageRequests.single()
+      request.channel.assert.isEqualTo(testData.slackConfig.channelId)
+    }
+  }
+
+  @Test
+  fun `Don't send a message if the subscription isn't global and modified language isn't in the preferred languages`() {
+    val testData = SlackTestData()
+    testDataService.saveTestData(testData.root)
+    mockedSlackClient = mockSlackClient()
+
+    val updatedConfig =
+      SlackConfigDto(
+        project = testData.projectBuilder.self,
+        slackId = "testSlackId",
+        channelId = "testChannel",
+        userAccount = testData.user,
+        languageTag = "fr",
+        onEvent = EventName.ALL,
+        slackTeamId = "",
+      )
+    slackConfigService.delete(testData.slackConfig.project.id, "testChannel")
+    val config = slackConfigService.createOrUpdate(updatedConfig)
+
+    loginAsUser(testData.user.username)
+
+    modifyTranslationData(testData.projectBuilder.self.id, "cs")
+    mockedSlackClient.chatPostMessageRequests.assert.hasSize(0)
```

Comment:
what do you mean by batch operation?

Reply:
Each time your `SlackSubscriptionProcessor.process` method is executed. There is a `io.tolgee.model.batch.BatchJob` entity created, which is asynchronously consumed by `BatchJobActionService`.

The problem is that you don'tknow if the `mockedSlackClient.chatPostMessageRequests.assert.hasSize(0)` is affected successfully because the batch operation didn't started yet or if it's because there is no batch operation to start.

But maybe testing it via the BatchOperations is not the right way either. Maybe the test is too complex and testing that the automation was removed would be enough.